src/control/lib/control/network.go

```diff
@@ -43,15 +43,22 @@ import (
 
 // HostFabricInterface describes a host fabric interface.
 type HostFabricInterface struct {
-	Provider string
-	Device   string
-	NumaNode uint32
+	Provider    string
+	Device      string
+	NumaNode    uint
+	Priority    uint
+	NetDevClass uint
```

Comment:
Out of curiosity, why the switch to uint? On 64-bit arch int/uint is 64 bits wide. If anything, we could probably make these uint8 if the goal is to be more efficient.

Reply:
just looked like less conversions this way, will change back to uint32

Reply:
also wasn't sure whether we should just be using the system types Rather than specifying fixed sizes

Reply:
Well, these are serialized to protobuf messages, which use uint32. That was why the type was chosen here. I would prefer to keep them as uint32 to avoid any possibility of weird overflow bugs. I can't see how any of these values would ever overflow a uint32, but that's the sort of bug that could cause a lot of frustration.

Reply:
done, left fabric config numa node type as uint because unpicking ended up messy  and contributed  to PR scope creep. so keeping as uint32 probably result in more type conversions , but it is consistent with most of the pb types